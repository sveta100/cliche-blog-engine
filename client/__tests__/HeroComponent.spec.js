import { shallowMount } from '@vue/test-utils';
import Hero from '../src/components/HeroComponent.vue';

describe('Hero component', () => {
  const $route = {
    path: '/',
  };

  const heroFactory = (router) => shallowMount(Hero, {
    mocks: {
      $route: router,
    },
  });

  test('renders correctly', () => {
    const wrapper = heroFactory($route);
    expect(wrapper.html()).toMatchSnapshot();
    const heroWrapper = wrapper.find('.hero-wrapper');
    expect(heroWrapper.exists()).toBe(true);
  });

  test('renders correct markup', () => {
    const wrapper = heroFactory($route);
    const mainTitle = wrapper.find('#mainTitle');

    expect(mainTitle.exists()).toBe(true);
  });

  test('does not show hero image if not on the main page', () => {
    const wrapper = heroFactory({
      path: '/about',
    });

    const mainTitle = wrapper.find('#main-title');
    const heroImage = wrapper.find('.hero');
    expect(mainTitle.exists()).toBe(false);
    expect(heroImage.exists()).toBe(false);
  });
});
