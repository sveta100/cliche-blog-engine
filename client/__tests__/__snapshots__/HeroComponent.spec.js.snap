// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component renders correctly 1`] = `
"<section class=\\"hero-wrapper mb-lg\\">
  <div class=\\"hero-wrapper hero-wrapper__content\\">
    <h1 id=\\"mainTitle\\">
      In Wonderland of Programming
    </h1>
    <div class=\\"hero_wrapper hero-wrapper__hero\\"></div>
    <div class=\\"hero-wrapper__decor-container\\"></div>
  </div>
</section>"
`;
